src/background/hypothesis-chrome-extension.js

```diff
@@ -178,14 +183,24 @@ export default function HypothesisChromeExtension({
   // Two important events are when the tab's `status` changes to `loading`
   // when the user begins a new navigation and when the tab's status changes
   // to `complete` after the user completes a navigation
-  function onTabUpdated(tabId, changeInfo, tab) {
-    if (changeInfo.status === TAB_STATUS_LOADING) {
-      resetTabState(tabId, tab.url);
-      const query = directLinkQuery(tab.url);
+  /**
+   *
+   * @param {number} tabId
+   * @param {chrome.tabs.TabChangeInfo} changeInfo
+   * @param {chrome.tabs.Tab} tab
+   */
+  function onTabUpdated(tabId, { status }, tab) {
+    const url = /** @type {string} - property included because manifest has the "tabs" permission*/ (tab.url);
+    const loadingUrl = isLoading.get(tabId);
+    if (status === 'loading' && url !== loadingUrl) {
+      isLoading.set(tabId, url);
+      resetTabState(tabId, url);
+      const query = directLinkQuery(url);
       if (query) {
-        state.setState(tab.id, { directLinkQuery: query });
+        state.setState(tabId, { directLinkQuery: query });
```

Comment:
@robertknight 

Do you think it could be a difference between `tabId` and `tab.id`?